portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes user comments. */
+@WebServlet("/delete-data")
+public class DeleteServlet extends HttpServlet {
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
As a thought exercise, how could you implement a servlet and front end to support deleting an individual comment? Say each comment has a `Delete` button or an `X`, and when you click it it deletes that individual comment.

Reply:
comment-form.js: 
-when forming comments, include an X icon within each paragraph:
  ~stylize it to appear as a button
  ~include an onclick() method that would use a query string to pass the id of the clicked-upon 
    comment to the servlet via POST
       +upon await obtaining a response from the servlet, I would then call my format feed 
         functions that I've already written--the specific comment should now be deleted and 
         should no longer show up after the page is refreshed

DeleteInvidualServlet.java:
-create a DatastoreService 
-create a doPost() function that parses the id from the query string (and type checks/ensures proper parameter formatting etc.)
  ~use KeyFactory to create a key for the parsed id 
  ~delete this specific key
  ~return an empty string

Reply:
Nice, that sounds good to me 👍